crates/host/src/secrets.rs

```diff
@@ -138,25 +139,27 @@ impl Manager {
             .then(|secret_name| async move {
                 match self.config_store.get(&secret_name).await {
                     Ok(Some(secret)) => serde_json::from_slice::<SecretReference>(&secret)
-                        .map_err(|_| anyhow::anyhow!("failed to deserialize secret reference from config store, ensure {secret_name} is a secret reference and not configuration")),
+                        .map_err(|_| anyhow::anyhow!("failed to deserialize secret reference from config store, ensure {secret_name} is a secret reference and not configuration"))
+                        .map(|secret_ref| (secret_name.trim_start_matches(SECRET_PREFIX).to_string(), secret_ref)),
                     Ok(None) => bail!(
                         "Secret reference {secret_name} not found in config store"
                     ),
                     Err(e) => bail!(e),
                 }
             })
             // Retrieve the actual secret from the secrets backend
-            .and_then(|secret_ref| async move {
+            .and_then(|(secret_name, secret_ref)| async move {
                 let secrets_client = self
                     .get_or_create_secrets_client(&secret_ref.backend)
                     .await?;
-                let application = Application{
+                let application = Application {
+                    // We pass an empty string if the entity doesn't belong to an application
                     name: application.cloned().unwrap_or_default(),
-                    policy: secret_ref.policy_properties.clone(),
+                    policy: secret_ref.policy,
                 };
```

Comment:
Is there any reason to not just inline this below inside of the `SecretRequest` like `Context` is already?

Reply:
I don't think so, let me see if it looks bad